delphi-checks/src/main/java/au/com/integradev/delphi/checks/VariableInitializationCheck.java

```diff
@@ -349,19 +351,21 @@ private static ExpressionNode skipHardCasts(ExpressionNode expression) {
       if (argumentList == null || !isHardCast(argumentList)) {
         return expression;
       }
-      expression = (ExpressionNode) argumentList.getChildren().get(0);
+      expression = argumentList.getArgumentNodes().get(0).getExpression();
     }
   }
 
   private static boolean isHardCast(ArgumentListNode argumentList) {
-    Node previous = argumentList.getParent().getChild(argumentList.getChildIndex() - 1);
-    if (previous instanceof NameReferenceNode) {
-      NameReferenceNode nameReference = ((NameReferenceNode) previous);
-      NameDeclaration declaration = nameReference.getLastName().getNameDeclaration();
-      return declaration instanceof TypeNameDeclaration;
-    } else if (previous instanceof CommonDelphiNode) {
-      DelphiTokenType tokenType = previous.getTokenType();
-      return tokenType == DelphiTokenType.STRING || tokenType == DelphiTokenType.FILE;
+    if (!argumentList.isEmpty()) {
+      Node previous = argumentList.getParent().getChild(argumentList.getChildIndex() - 1);
+      if (previous instanceof NameReferenceNode) {
+        NameReferenceNode nameReference = ((NameReferenceNode) previous);
+        NameDeclaration declaration = nameReference.getLastName().getNameDeclaration();
+        return declaration instanceof TypeNameDeclaration;
+      } else if (previous instanceof CommonDelphiNode) {
+        DelphiTokenType tokenType = previous.getTokenType();
+        return tokenType == DelphiTokenType.STRING || tokenType == DelphiTokenType.FILE;
+      }
```

Comment:
This code seems to duplicate the behaviour of `CastUtils.readHardCast` - could we replace this with a call to that method?

Reply:
On the surface I agree - as soon as I tried, it turned into a rather annoying spiderweb of static methods all calling each other and requiring us to pass `DelphiCheckContext` through all of them in order to get the `TypeFactory` all the way through to the `CastUtils.readHardCast` call.

Of course, we don't even need the type factory here because we don't care what the cast types are.

Anyway, let's put a pin in this one. I don't like the idea of a whole `VariableInitizationCheck` refactor in this PR.

Reply:
Okay, sure. I think this should be done the next time `VariableInitializationCheck` is changed - the reason for adding `CastUtils` in the first place was to prevent this sort of complex code duplication.